community-spotlight-conference-backend-world-affairs-conference.mdx

```diff
@@ -0,0 +1,92 @@
+
+For the backend, there were a few very specific requirements for different parts of the registration flows, administrative dashboards and general budget. The backend needed to:
+
+- Be able to host images
+- Speak SQL and support relational data models
+- Be a fully managed and hosted solution
+- Be cost effective and fall within the constraints of the conference budget
+
+Xata checked all of the boxes and more for their next generation tech stack.
+
+> I think we used pretty much every feature Xata offers. Our user profile and speaker images are stored in Xata, we used the summarize API for sign up and registration time series charts, preview databases with Netlify, full text search for our application and more. Having one service to provide all of this functionality made it a pretty easy decision to choose Xata.\
+> _Samuel Martineau - Title_
+
+On the frontend, [Sveltekit](https://kit.svelte.dev/), [three.js](https://threejs.org/) and [AG Grid](https://ag-grid.com/) were used to build the website, registration application and internal administrative dashboards. All of this was deployed on [Netlify](https://www.netlify.com/), where the [Xata - Netlify](https://xata.io/docs/integrations/netlify) integration was used to provide preview deployments in their CI/CD pipeline. The pull request generated previews were critical to the development process for the team, making it easy to test changes before deploying to production.
+
+## A registration flow with optimized session assignment
+
+As registration flows should be, this one was pretty simple overall and leveraged [Lucia Auth](https://lucia-auth.com/) to provide a secure onboarding experience. The form intakes your personal details, as well as lunch preferences and plenary sessions you’re most interested in attending. On submit, the data is then posted to the Xata database and the user is prompted with a security code to confirm their email and identity.
+
+![Input form for conference registration](/images/cs-wac-registration-form.png)
+
+In most cases, this would be the end of the registration flow. Because this is a conference with some pretty popular and well known speakers, Samuel needed to come up with a way to provide the most optimal schedule to attendees. How can they provide personalized conference tracks that are optimized to make the most people happy?
+
+It turns out that the problem of attendee assignment to plenaries is very similar to the problem faced when assigning medical residents to hospitals for their medication. The WAC team ended up using the same algorithm to solve this problem, the [Gale-Shapely algorithm](https://en.wikipedia.org/wiki/Gale%E2%80%93Shapley_algorithm). They created a weighted graph between every single attendee and every single plenary. The weight of the edges are stored in Xata and Samuel and his team created an algorithm to provide an optimal assignment of sessions for every single attendee.
```

Comment:
In the context to the Gale-Shapely algorithm, "medical residents" refers not to patients, but to students in the process of becoming doctors, so the part about medication should be removed.

Reply:
Updated ✅